backend/pkg/transport/network/tcp/tcp_test.go

```diff
@@ -0,0 +1,86 @@
+package tcp_test
+
+import (
+	"encoding/json"
+	"github.com/HyperloopUPV-H8/h9-backend/pkg/transport/network/tcp"
+	"github.com/rs/zerolog"
+	"os"
+	"testing"
+	"time"
+)
+
+type Socket struct {
+	SrcIP   string
+	SrcPort uint16
+	DstIP   string
+	DstPort uint16
+}
+
+// Payload defines a piece of information comming from the network with its metadata
+type Payload struct {
+	Timestamp time.Time
+	Socket    Socket
+	Data      []byte
+}
+
+func TestTCP(t *testing.T) {
+	writer := os.Stdout
+	logger := zerolog.New(writer)
+
+	// Server setup
+	serverAddr := "127.0.0.1:3000"
+	server := tcp.NewServer(serverAddr, tcp.ServerConfig{}, logger)
+	go func() {
+		if err := server.Listen(); err != nil {
+			t.Fatalf("Server failed to listen: %v", err)
+		}
+	}()
+	time.Sleep(1 * time.Second)
```

Comment:
avoid `time.Sleep` in tests, in this case you don't need to sleep, if the server is not ready the client should will wait a few milliseconds (based on the backoff algorithm) to try again, and the default config does this forever